Flaky CI tests: test_truncation_of_waveform and test_cropping_bounds_independence fail stochastically

## Description
Two tests fail intermittently in CI due to stochastic behavior (no fixed random seed). The failures appeared in PR #375 which added a multi-Python-version matrix, but the flakiness is not version-specific — different Python versions fail on different runs, and 3.11/3.13 passed in the same run that failed on 3.10/3.12.

Failing run: https://github.com/dingo-gw/dingo/actions/runs/24454582083

## Failures Observed

### Python 3.10 (job 71451808480)
`tests/gw/test_waveform_dataset.py::test_truncation_of_waveform`
```
>           assert np.allclose(b / scale_factor, a / scale_factor)
E           assert False
tests/gw/test_waveform_dataset.py:167: AssertionError
```

### Python 3.12 (job 71451808483)
`tests/gw/transforms/test_waveform_transforms.py::test_cropping_bounds_independence`
```
FAILED ... - assert (np.float64(0.0) > 0) == True
```
Triggered at line 248: `assert (mask.std(axis=-3).max() > 0) == independent_detectors`

## Root Cause (suspected)
- `test_cropping_bounds_independence` uses `CropMaskStrainRandom` with `cropping_probability=0.8` on a single (non-batched) sample. With probability 0.2 no cropping occurs, so `mask` is all `False` and `mask.std().max() == 0`, failing the `== True` branch when `independent_detectors=True`.
- The fixture in `tests/gw/transforms/test_waveform_transforms.py:16,35` uses `np.random.normal(...)` without a fixed seed.
- `test_truncation_of_waveform` in `tests/gw/test_waveform_dataset.py:137` also depends on waveform-generation fixtures that do not appear to be deterministically seeded.

## Expected Behavior
Tests pass reliably on every Python version.

## Suggested Fixes
- Seed `np.random` (or use `np.random.default_rng(seed)`) in the fixtures at `tests/gw/transforms/test_waveform_transforms.py:16` and `:35`.
- For `test_cropping_bounds_independence`: either seed the RNG inside the transform test, raise the effective sample count so the probability of zero crops is negligible, or explicitly retry / force cropping on the single-sample path.
- Investigate whether `test_truncation_of_waveform` (marked `@pytest.mark.slow`) has a similar unseeded source of variation in the generated waveform dataset.

## Environment
- Python: 3.10 and 3.12 (3.11, 3.13 passed in the same matrix run)
- OS: ubuntu-latest (GitHub Actions)
- Commit: PR #375 head

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -1,0 +1,15 @@
+import bilby.core.utils.random
+import numpy as np
+import pytest
+
+
+@pytest.fixture(autouse=True)
+def _seed_rngs():
+    """Seed numpy and bilby RNGs before every test.
+
+    Makes the full suite deterministic: each test starts from the same known
+    state, independent of test ordering. Avoids stochastic failures like the
+    ones tracked in issue #377.
+    """
+    np.random.seed(0)
+    bilby.core.utils.random.seed(0)

--- a/tests/gw/transforms/test_waveform_transforms.py
+++ b/tests/gw/transforms/test_waveform_transforms.py
@@ -220,11 +220,14 @@ def test_cropping_bounds_independence(cropping_setup):
     example_batch_single = dict(waveform=strain_in[0])
 
     for independent_detectors in [False, True]:
+        # cropping_probability=1.0: the probabilistic skip is covered by
+        # test_cropping_probability; here we only assert bounds independence,
+        # which is unambiguous when every sample is cropped.
         cropping_transform = CropMaskStrainRandom(
             domain,
             f_min_upper=32,
             f_max_lower=512,
-            cropping_probability=0.8,
+            cropping_probability=1.0,
             independent_detectors=independent_detectors,
             independent_lower_upper=False,
         )

